e2-studio-star-rx72n-firmware/libs/rx_spi_link/src/rx_spi_link.c

```
@@ -0,0 +1,731 @@
+    return k_rx_err_invalid_size;
+  }
+
+  /* Reset HARQ for new transaction */
+  rx_err_t err = rx_harq_reset(&link->harq);
+  if (err != k_rx_ok) {
+    rx_log_error(s_tag, "HARQ reset failed");
+    return err;
+  }
+
+  /* Prepare payload: FEC encode if enabled, otherwise passthrough */
+  const uint8_t* tx_payload = payload;
```

Clang (14.0.6)

[warning] variable 'tx_payload' is not initialized

(cppcoreguidelines-init-variables)